### PR TITLE
Update mass fluxes at end of timestep with full timestep's worth

### PR DESCRIPTION
**Pull request by maxpkatz**
## PR summary

This update makes the mass_fluxes array hold the full timestep's worth of mass fluxes at the end of finalize_advance. (This subtlety only matters if there were multiple subcycles in the timestep, in which case mass_fluxes would presently only have the last subcycle's worth of mass fluxes.)

## PR motivation

This is a first step towards implementing the request in #1691.

## PR checklist

- [x] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
tests pass: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2022-04-28-005/index.html

File: Source/driver/Castro_advance.cpp [hunk 1]
```diff
@@ -501,6 +501,16 @@ Castro::finalize_advance()
         FluxRegFineAdd();
     }
 
+    // The mass_fluxes array currently holds only the fluxes
+    // from the last subcycle. Override this with the sum of
+    // the fluxes from the full timestep (this will be used
+    // later during the reflux operation).
+
```
Comment (zingale) [inline Source/driver/Castro_advance.cpp:508]: the PR says this change makes mass_fluxes hold the full timestep's worth of fluxes -- doesn't this comment say the opposite?

Comment (maxpkatz) [reply Source/driver/Castro_advance.cpp:508]: It is assuming that `fluxes`, which is about to be copied to `mass_fluxes`, has the full timestep's worth of flux data at this point.

Comment (zingale) [reply Source/driver/Castro_advance.cpp:508]: ah, I understand.

**Review by zingale — APPROVED** [review]

**Comment by zingale** [comment]
tests pass: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2022-10-27-001/index.html

**Pull request merged**